### PR TITLE
feat(Instructor): integrate with SQLModel, provide documentation and example

### PR DESCRIPTION
**Pull request by shanktt**
| <a href="https://ellipsis.dev" target="_blank">Ellipsis</a> | :rocket: This PR description was created by [Ellipsis](https://www.ellipsis.dev) for commit eab54e63dc56008c9486c4ba7cb8c40b99beb89b.  | 
|--------|--------|

### Summary:
This PR integrates Instructor with SQLModel, providing documentation and an example on how to define models, generate records using OpenAI, and insert the responses into a database.

**Key points**:
- Added a new documentation file `/docs/concepts/sqlmodel.md` explaining the integration of Instructor with SQLModel.
- Defined a `Hero` model in both the documentation and the example.
- Demonstrated how to generate a `Hero` record using OpenAI in the `create_hero` function.
- Showed how to insert the generated record into a database.
- Included a runnable example in `/examples/sqlmodel/run.py` demonstrating the process.
----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

**Comment by jxnl** [comment]
How are you liking ...

**Comment by jxnl** [comment]
Can you move this to examples / cookbooks for now.

**Comment by shanktt** [comment]
> How are you liking ...

Definitely better than what I'd write myself lol

File: mkdocs.yml [hunk 1]
```diff
@@ -166,6 +166,7 @@ nav:
     - Enabling Open Source Models: 'examples/open_source.md'
     - Image to Ad Copy: 'examples/image_to_ad_copy.md'
     - Ollama + LiteLLM: 'examples/ollama.md'
```
Comment (jxnl) [inline mkdocs.yml:168]:
```suggestion
    - Ollama: 'examples/ollama.md'
```

File: examples/sqlmodel/run.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import instructor
+from openai import OpenAI
+from typing import Optional
+from sqlmodel import Field, SQLModel, create_engine, Session
+
+
+# Define the model that will serve as a Table for the database
+class Hero(SQLModel, instructor.OpenAISchema, table=True):
```
Comment (jxnl) [inline examples/sqlmodel/run.py:8]: we need to name this OpenAISchema its so long.

**Pull request merged**